File: 01-two-sum/solution.ts
function twoSum(nums: number[], target: number): number[] {
    const mp = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const rem = target - nums[i];

        if (mp.has(rem)) {
            return [i, mp.get(rem)!];
        }

        mp.set(nums[i], i);
    }

    return [];
}

File: 01-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = {}

        for i in range(len(nums)):
            rem = target - nums[i]

            if rem in mp:
                return [i, mp[rem]]

            mp[nums[i]] = i

        return []